### PR TITLE
Enhancement/65 migration e2e test

### LINKED ISSUE #65: Testing | Add E2E test for migration script
Create E2E tests to verify that the migration script (run when upgrading from v4, to move settings from the local storage into the Chrome's storage) works correctly.

First test:

1. Populate the local store with sample data (which should be different than default data). Hint: maybe this can be done on the `options.html` screen.
2. Open the `migration.html` screen and check that user is redirected to the `INSTALLED_URL`.
3. Verify than data from local store has been copied into Chrome's storage.
4. Verify than values don't defined in local store (like `mergeSearchProviders` which was added on v5) are defined.

Second test:

1. Clear the local store (should be empty).
2. Open the `migration.html` screen and check that user is redirected to the `INSTALLED_URL`.
3. Verify than data on Chrome's storage has been populated using the values from `settings.json`.

Blocked by #59, related to #57

### PR DESCRIPTION
**Pull request by tim-kodirov**
Fixes #65 

To be merged after https://github.com/AdvancedThreatAnalytics/threat-analytics-search/pull/61

### CODE REVIEW

File: tests/e2e/migration.spec.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  });
+
+  test("Data from local storage is copied to chrome's storage", async () => {
+    // Go to options page to set sample data to local storage
+    page = await ExtensionUtil.goto("options");
+
+    const sampleData = {
+      key: "_configUrl",
+      value: "test value",
+    };
```
Comment (jfmdev) [inline tests/e2e/migration.spec.js:22]: I should have been more clear on the issue, but here you are just checking that `_configUrl` is being migrated, and you should check that **all** values (used on v4) are migrated (e.g. `_configLastRefresh`, `_configUseGroups`, `_configEnc`, `_configAutoRefresh`, `_group1Name`, `_allsearch`, `_RSAConfig`, `_RSAallquery`, and so on)

File: tests/e2e/migration.spec.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    // Get data from settings.json
+    const defaultProviders = ConfigFile.parseProviders(
+      DefaultFile.searchproviders
+    );
+
+    // Verify if chrome's storage is populated from settings.json
+    expect(_.map(providers, (item) => _.omit(item, "menuIndex"))).toEqual(
+      _.map(defaultProviders, (item) => _.omit(item, "menuIndex"))
+    );
```
Comment (jfmdev) [inline tests/e2e/migration.spec.js:93]: Besides verifying that `SETTINGS` and `SEARCH_PROVIDERS` have been populated, you should also check `CARBON_BLACK`, `NET_WITNESS` and `RSA_SECURITY`,

**Review by jfmdev — CHANGES_REQUESTED** [review]

File: tests/e2e/migration.spec.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+      StoreKey.SETTINGS,
+      StoreKey.SEARCH_PROVIDERS
+    );
+
+    // Verify if data from local storage is copied to chrome storage
+    expect(sampleValue.configurationURL).toEqual(sampleData.value);
+
+    // Verify if data not defined in local storage is in chrome storage
+    expect(providerValue).toBeTruthy();
+  });
```
Comment (jfmdev) [inline tests/e2e/migration.spec.js:57]: Also `Verify than values don't defined in local store (like mergeSearchProviders which was added on v5) are defined.`

**Review by jfmdev — COMMENTED** [review]
I'm getting errors when running the tests:

image

**Comment by Saidmurod1996** [comment]
I've tested multiple times. Now all tests pass

**Comment by tim-kodirov** [comment]
Tests are running successfully, but warning messages are shown there:

image

**Review by jfmdev — APPROVED** [review]

**Pull request merged**